nntrainer/layers/layer_node.h

```diff
@@ -739,15 +625,15 @@ class LayerNode final : public ml::train::Layer, public GraphNode {
    * @details this is layer inside the distribution layer if this layer node
    * is distributed.
    */
-  const std::shared_ptr<nntrainer::LayerV1> &getLayer() const;
+  const nntrainer::Layer *getLayer() const;
```

Comment:
Is there any reason to use pointer instead of shared_pointer?

Reply:
internal layer has not become a unique_ptr<>, however, unique_ptr<> cannot be returned.
for unique_ptr, in order to return the object to non-owners, the underlying ptr is returned.